### PR TITLE
Max/position apply

### PR DESCRIPTION
**Pull request by MengxuanChen**

### CODE REVIEW

File: src/components/PositionList.tsx [hunk 1]
```diff
@@ -17,6 +17,11 @@ const PositionList: FC<PLProps> = ({ election }) => {
 
   const showDelete = user.user.id === election.manager;
 
+  //Navigate to the position application form
+  function navToPositionApplyForm(positionId:string): void{
+    history.push(`/apply/${positionId}`)
+  }
+
```
Comment (RyanFleck) [inline src/components/PositionList.tsx:24]: Could you integrate this into the `onClick` on line 59? I'd say this breakout function is not neccessary for a single call.

File: src/pages/PositionApply.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if(!ctx || !ctx.credentials.authenticated) return;
+    //If logged in, attempt to get the position details
+    getPosition(positionId)
+        .then((res) => {
+            setPosition(res);
+        })
+        .catch((err)=>{
+            console.error(err)
+        })
+  },[]);
```
Comment (RyanFleck) [inline src/pages/PositionApply.tsx:42]: This should be sensitive to `positionId`

File: src/utils/api/PositionApplication.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      ): Promise<Position>{
+        const token = await preRequestRefreshAuth();
+        return api
+        .get(positionManagementUrl+positionId,{
+            headers: {Authorization: `JWT ${token}`}
+        })
+        .then((res)=>{
+            const p: Position = res.data;
+            return p;
+        })
```
Comment (RyanFleck) [inline src/utils/api/PositionApplication.ts:37]: **Perfect**

File: src/utils/api/PositionApplication.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  export type PositionApplicationParams = {
+      platform: string;
+      position: string;
+  };
+
+  const positionManagementUrl = `/elections/manage/position/`;
+  const positionApplicationSubmitUrl = `/elections/participate/candidate/`
+  export async function getPosition(
+      positionId:string
+      ): Promise<Position>{
```
Comment (RyanFleck) [inline src/utils/api/PositionApplication.ts:28]: Excellent.

File: src/utils/api/PositionApplication.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    const token = await preRequestRefreshAuth();
+    return api
+        .post(positionApplicationSubmitUrl, formData, {
+            headers: {Authorization: `JWT ${token}`},
+        })
+        .then((res) => {
+            const c: Candidate = res.data;
+            return c;
+        })
+  }
```
Comment (RyanFleck) [inline src/utils/api/PositionApplication.ts:52]: Make sure to run `yarn format` before you request review again.

**Review by RyanFleck — APPROVED** [review]
A few small changes, otherwise looks good. Will check out visually tomorrow and merge.

**Comment by RyanFleck** [comment]
image

Might also be good to add some `<br />` between the title/description title and description/application title

**Review by RyanFleck — CHANGES_REQUESTED** [review]
A few small changes/tweaks, but great job for now.

I'll make that public position endpoint soon.

File: src/utils/api/PositionApplication.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    position: string;
+    platform: string;
+  };
+
+  export type PositionApplicationParams = {
+      platform: string;
+      position: string;
+  };
+
+  const positionManagementUrl = `/elections/manage/position/`;
```
Comment (RyanFleck) [inline src/utils/api/PositionApplication.ts:24]: This URL needs to be changed to the one specified in the ticket.

File: src/pages/PositionApply.tsx (hunk 2 above)
Comment (MengxuanChen) [reply src/pages/PositionApply.tsx:42]: wym

Comment (RyanFleck) [reply src/pages/PositionApply.tsx]:
```
  },[]);
```

What does this do?

Comment (MengxuanChen) [reply src/pages/PositionApply.tsx]: as in put id in the [ ]

File: src/components/PositionList.tsx [hunk 7]
```diff
@@ -17,6 +17,11 @@ const PositionList: FC<PLProps> = ({ election }) => {
 
   const showDelete = user.user.id === election.manager;
 
+  //Navigate to the position application form
+  function navToPositionApplyForm(positionId: string): void {
+    history.push(`/apply/${positionId}`);
+  }
+
```
Comment (RyanFleck) [inline src/components/PositionList.tsx:24]: Remove this function, lol.

**Review by RyanFleck — CHANGES_REQUESTED** [review]
One more tweak.

File: src/pages/PositionApply.tsx (hunk 2 above)
Comment (MengxuanChen) [reply src/pages/PositionApply.tsx]: The effect gonna execute every time the variable in the [] changes

**Pull request merged**